canopen/source/co_sync.c

```diff
@@ -182,30 +186,125 @@ void COSyncRestart(CO_SYNC *sync)
 /*
 * see function definition
 */
-CO_ERR COTypeSyncIdWrite(struct CO_OBJ_T *obj, struct CO_NODE_T *node, void *buf, uint32_t len)
-{
-    CO_ERR    result = CO_ERR_NONE;
-    uint32_t  nid;
-    uint32_t  oid;
+void COSyncProdActivate(CO_SYNC *sync) {
+    uint32_t ticks, time;
+    CO_ERR   err;
+    CO_NODE *node;
+    int16_t  tid;
+
+    node = sync->Node;
+
+    err = CODictRdLong(&node->Dict, CO_DEV(0x1006, 0), &sync->Cycle);
+    if (err != CO_ERR_NONE) {
+        node->Error = CO_ERR_CFG_1006_0;
```

Comment:
This error forces all users to use the stack with the SYNC Producer feature. As this object entry is:
> Conditional;
> Mandatory, if PDO communication on a synchronous base is supported

We can ignore this error (and set `sync->Cycle` to 0). This means: we accept a system without SYNC producer.

Reply:
Good idea! This way we have conditional & mandatory checked.